src/test/java/org/broadinstitute/hellbender/engine/MultiVariantDataSourceUnitTest.java

```diff
@@ -0,0 +1,145 @@
+    public void testQueryOverUnindexedFile() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+        multiVariantSource.addFeatureDataSource(new File(ENGINE_TEST_DIRECTORY + "unindexed.vcf"), "unindexed");
+        multiVariantSource.query(new SimpleInterval("1", 1, 1));
+    }
+
+    @Test
+    public void testGetName() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+
+        multiVariantSource.addFeatureDataSource(baseVariants, "sourceName1");
+        String name = multiVariantSource.getName();
+        Assert.assertTrue(name.contains("sourceName1"));
+
+        multiVariantSource.addFeatureDataSource(baseVariantsAlternateDictionary, "sourceName2");
+        name = multiVariantSource.getName();
+        Assert.assertTrue(name.contains("sourceName1"));
+        Assert.assertTrue(name.contains("sourceName2"));
+    }
+
+    @Test
+    public void testGetSequenceDictionaryCompatible() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+        multiVariantSource.addFeatureDataSource(new File(MULTI_VARIANT_TEST_DIRECTORY, "interleavedVariants_1.vcf"), "interleavedVariants_1");
+        multiVariantSource.addFeatureDataSource(new File(MULTI_VARIANT_TEST_DIRECTORY, "interleavedVariants_2.vcf"), "interleavedVariants_2");
+
+        Assert.assertEquals(multiVariantSource.getSequenceDictionary().getSequences().size(), 4);
+    }
+
+    @Test
+    public void testGetSequenceDictionaryAlternate() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+        multiVariantSource.addFeatureDataSource(baseVariants, "baseVariants");
+        multiVariantSource.addFeatureDataSource(baseVariantsAlternateDictionary, "baseVariantsAlternateDictionary");
+
+        multiVariantSource.getSequenceDictionary();
+    }
+
+    @Test(expectedExceptions = UserException.IncompatibleSequenceDictionaries.class)
+    public void testGetSequenceDictionaryIncompatible() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+        multiVariantSource.addFeatureDataSource(baseVariants, "baseVariants");
+        multiVariantSource.addFeatureDataSource(baseVariantsConflictingDictionary, "baseVariantsConflictingDictionary");
+
+        multiVariantSource.getSequenceDictionary();
+    }
+
+    @Test
+    public void testIterator() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+        multiVariantSource.addFeatureDataSource(new File(MULTI_VARIANT_TEST_DIRECTORY, "interleavedVariants_1.vcf"), "interleavedVariants_1.vcf");
+        multiVariantSource.addFeatureDataSource(new File(MULTI_VARIANT_TEST_DIRECTORY, "interleavedVariants_2.vcf"), "interleavedVariants_2.vcf");
+
+        int count = 0;
+        for (final VariantContext vc: multiVariantSource) {
+            count++;
+        };
+        Assert.assertEquals(count, 26);
+    }
```

Comment:
Add a test case in which you make assertions about the order in which variants are returned, like in `FeatureDataSourceUnitTest`